src/Sulu/Bundle/LocationBundle/Resources/js/containers/Location/LocationOverlay.js

```diff
@@ -0,0 +1,261 @@
+    handleConfirm = () => {
+        const {title, street, number, code, town, country, markerLat, markerLong, mapZoom} = this;
+
+        if (!markerLat || !markerLong) {
+            this.props.onConfirm(null);
+
+            return;
+        }
+
+        this.props.onConfirm({
+            title,
+            street,
+            number,
+            code,
+            town,
+            country,
+            lat: markerLat,
+            long: markerLong,
+            zoom: mapZoom,
+        });
+    };
+
+    @action handleAutoCompleteChange = (data: Object) => {
+        console.log('select', toJS(data));
+
+        this.mapLat = data.latitude || 0;
+        this.mapLong = data.longitude || 0;
+
+        this.markerLat = data.latitude;
+        this.markerLong = data.longitude;
+
+        this.title = data.displayTitle;
+        this.street = data.street;
+        this.number = data.number;
+        this.code = data.code;
+        this.town = data.town;
+        this.country = data.country;
+    };
+
+    @action handleMapZoom = (event: Object) => {
+        this.mapZoom = event.zoom;
+    };
+
+    @action handleMarkerDrag = (event: Object) => {
+        this.markerLong = event.latlng.lng;
+        this.markerLat = event.latlng.lat;
+    };
+
+    @action handleMarkerDragEnd = () => {
+        this.mapLong = this.markerLong || 0;
+        this.mapLat = this.markerLat || 0;
+    };
+
+    @action handleResetLocation = () => {
+        this.mapLat = 0;
+        this.mapLong = 0;
+        this.mapZoom = 1;
+
+        this.markerLong = null;
+        this.markerLat = null;
+
+        this.title = null;
+        this.street = null;
+        this.number = null;
+        this.code = null;
+        this.town = null;
+    };
+
+    @action handleTitleChange = (title: ?string) => {
+        this.title = title;
+    };
+
+    @action handleStreetChange = (street: ?string) => {
+        this.street = street;
+    };
+
+    @action handleNumberChange = (number: ?string) => {
+        this.number = number;
+    };
+
+    @action handleCodeChange = (code: ?string) => {
+        this.code = code;
+    };
+
+    @action handleTownChange = (town: ?string) => {
+        this.town = town;
+    };
+
+    @action handleCountryChange = (country: ?string) => {
+        this.country = country;
+    };
+
+    @action handleLatChange = (lat: ?number) => {
+        this.mapLat = lat || 0;
+        this.markerLat = lat;
+    };
+
+    @action handleLongChange = (long: ?number) => {
+        this.mapLong = long || 0;
+        this.markerLong = long;
+    };
+
+    @action handleZoomChange = (zoom: ?number) => {
+        this.mapZoom = zoom || 1;
+    };
+
+    render() {
+        const {
+            onClose,
+            open,
+        } = this.props;
+
+        // enable confirm button if all marker properties are set or no property is set in case of a reset
+        const confirmEnabled = (this.markerLat && this.markerLong) || (!this.markerLat && !this.markerLong);
```

Comment:
Not sure about this, haven't we handled this in other overlays in a different way? I would say we should only enable the confirm button if something has changed.

Reply:
i would like to enable the confirm button only if the overlay contains consistent/valid values. we use something similar in the `ExternalLinkOverlay` o the `FocusPointOverlay`. but i could add an additional `dirty` observable/computed if you want to check that too 🙂